packages/discovery/src/discovery/handlers/user/StateFromEvent.ts

```diff
@@ -0,0 +1,123 @@
+import { ContractValue, EthereumAddress } from '@l2beat/shared'
+import { utils } from 'ethers'
+import * as z from 'zod'
+
+import { DiscoveryLogger } from '../../DiscoveryLogger'
+import { DiscoveryProvider } from '../../provider/DiscoveryProvider'
+import { Handler, HandlerResult } from '../Handler'
+import { getEventFragment } from '../utils/getEventFragment'
+import { toContractValue } from '../utils/toContractValue'
+
+export type StateFromEventDefinition = z.infer<typeof StateFromEventDefinition>
+export const StateFromEventDefinition = z.strictObject({
+  type: z.literal('stateFromEvent'),
+  event: z.string(),
+  eventKeys: z.array(z.string()),
```

Comment:
Ah well, to be honest I got events mixed up with function overloading. So I thought that one could have:
```
event MyEvent(uint256 indexed id);
event MyEvent(uint256 indexed id, string name);
```

But that actually doesn't work. So the whole eventKeys is I guess pointless. I want the handler to be active whenever we see the event. If we do filter keys - it would mean that we want to extract only part of the event arguments.

event: "MyEvent"
filterKeys: ['id']

This will only then return the id, but then we have also the case where no filterKeys is provided - it would then return all the arguments.

Reply:
No, I'll add them. I think it is nice. You could have events that return booleans or bytes or other stuff we are not interested in.

Reply:
I want for returnParams, as I want the event params in the return object. I didn't like `filterKeys` for this. If you don't like `returnParams` I can live with `onlyKeys`.